src/consumer/__tests__/consumeMessages.spec.js

```diff
@@ -177,6 +178,139 @@ describe('Consumer', () => {
     ])
   })
 
+  testIfKafka011('consume messages with 0.11 format', async () => {
+    const topicName2 = `test-topic2-${secureRandom()}`
+    await createTopic({ topic: topicName2 })
+
+    cluster = createCluster({ allowExperimentalV011: true })
+    producer = createProducer({
+      cluster,
+      createPartitioner: createModPartitioner,
+      logger: newLogger(),
+    })
+
+    consumer = createConsumer({
+      cluster,
+      groupId,
+      maxWaitTimeInMs: 100,
+      logger: newLogger(),
+    })
+
+    jest.spyOn(cluster, 'refreshMetadataIfNecessary')
+
+    await consumer.connect()
+    await producer.connect()
+    await consumer.subscribe({ topic: topicName, fromBeginning: true })
+    await consumer.subscribe({ topic: topicName2, fromBeginning: true })
+
+    const messagesConsumed = []
+    await consumer.run({ eachMessage: async event => messagesConsumed.push(event) })
+
+    const generateMessages = () =>
+      Array(103)
+        .fill()
+        .map(() => {
+          const value = secureRandom()
+          return {
+            key: `key-${value}`,
+            value: `value-${value}`,
+            headers: {
+              'header-keyA': `header-valueA-${value}`,
+              'header-keyB': `header-valueB-${value}`,
+              'header-keyC': `header-valueC-${value}`,
+            },
+          }
+        })
+
+    const messages1 = generateMessages()
+    const messages2 = generateMessages()
+
+    await producer.sendBatch({
+      acks: 1,
+      topicMessages: [
+        { topic: topicName, messages: messages1 },
+        { topic: topicName2, messages: messages2 },
+      ],
+    })
+
+    await waitForMessages(messagesConsumed, { number: messages1.length + messages2.length })
+
+    expect(cluster.refreshMetadataIfNecessary).toHaveBeenCalled()
```

Comment:
Why is this check being done?

Reply:
The other tests also have the check, it's just to make sure that the function is called